### PR TITLE
add image-name flag to deploy command

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

Make it so you can provide an --image-name flag directly to the deploy command. This command will make it so a user can pass custom image to the deploy command built from the dockerfile

## 🎟 Issue(s)

- https://github.com/astronomer/astro-cli/issues/619
- https://github.com/astronomer/astro-cli/issues/627

## 🧪 Functional Testing

unit tests 

## 📸 Screenshots

deploy
Screen Shot 2022-06-29 at 10 48 12 AM

dev start
Screen Shot 2022-06-29 at 1 44 19 PM

## 📋 Checklist

- [X] Rebased from the main (or release if patching) branch (before testing)
- [X] Ran `make test` before taking out of draft
- [X] Ran `make lint` before taking out of draft
- [X] Added/updated applicable tests
- [X] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [X] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: airflow/docker_image.go [hunk 1]
```diff
@@ -206,6 +214,8 @@ var cmdExec = func(cmd string, stdout, stderr io.Writer, args ...string) error {
 	return nil
 }
 
+var CmdExec = cmdExec
```
Comment (sunkickr) [inline airflow/docker_image.go:217]: Was using this to test will delete

File: cmd/cloud/deploy_test.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ func TestDeployImage(t *testing.T) {
 		return nil
 	}
 
-	deployImage = func(path, deploymentID, wsID, pytest, envFile string, prompt bool, client astro.Client) error {
+	deployImage = func(path, deploymentID, wsID, pytest, envFile string, imageName string, prompt bool, client astro.Client) error {
```
Comment (kushalmalani) [inline cmd/cloud/deploy_test.go:26]: Should this change be?

```suggestion
	deployImage = func(path, deploymentID, wsID, pytest, envFile, imageName string, prompt bool, client astro.Client) error {
```

**Review by kushalmalani — APPROVED** [review]
Left a minor comment

**Review by neel-astro — APPROVED** [review]
looks fine, have left a minor comment

**Review by neel-astro — APPROVED** [review]
would be great if we can up the code coverage 😄

File: airflow/container.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ type ImageHandler interface {
 	Push(registry, username, token, remoteImage string) error
 	GetLabel(labelName string) (string, error)
 	ListLabels() (map[string]string, error)
+	RenameLocalImage(localImage string) error
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply airflow/container.go:42]: 👍

File: cmd/cloud/deploy_test.go (hunk 2 above)
Comment (sunkickr) [reply cmd/cloud/deploy_test.go:26]: yes thanks

File: airflow/container.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sunkickr) [reply airflow/container.go:42]: okay I'll rename the function

**Comment by sunkickr** [comment]
Made changes from code review

**Pull request merged**